examples/pubsub.rs

```diff
@@ -0,0 +1,27 @@
+extern crate env_logger;
+extern crate web3;
+
+use web3::futures::{Future, Stream};
+
+fn main() {
+    env_logger::init();
+
+    let (_eloop, ws) = web3::transports::WebSocket::new("ws://localhost:8546").unwrap();
+    let web3 = web3::Web3::new(ws.clone());
+    let mut sub = web3.eth_subscribe().subscribe_new_heads().wait().unwrap();
+
+    println!("Got subscription id: {:?}", &sub);
+
+    (&mut sub)
```

Comment:
Part of the reason for exposing subscription IDs in the initial PR was because this type of interface didn't seem very ergonomic to me but seemed to be needed to make the borrow checker happy. If there is a better way to do this let me know.

I think this won't be a problem with the upcoming futures 0.2 as for_each returns a future which resolves to the original stream, allowing you to call methods on it to clean stuff up, in the current version it resolves to unit.